helpers.js

```diff
@@ -0,0 +1,146 @@
+const fs = require('fs/promises');
+const prettier = require('prettier');
+const { relationshipTypes, widgetTypes } = require('./config');
+
+function handleFieldType(field) {
+  let newField;
+
+  if (field.type === 'tags' || field.type === 'array') {
+    newField = handleArray(field);
+  } else if (field.type === 'area') {
+    newField = handleArea(field);
+  } else if (relationshipTypes[field.type]) {
+    newField = handleRelationship(field);
+  } else {
+    newField = field;
+  }
+
+  return newField;
+}
+
+function handleArray (field) {
+  const schema =
+      field.type === 'tags'
+        ? [
+          {
+            ...field,
+            name: 'tag',
+            type: 'string',
+            label: 'Tag'
+          }
+        ]
+        : field.schema;
+
+  const arrayFields = schema.reduce(
+    (arrayAcc, arrayCur) => {
+      const {
+        name, moduleName, ...arrayProps
+      } = arrayCur;
+
+      const newArrayProps = handleFieldType(arrayProps);
+
+      arrayAcc.add[name] = newArrayProps;
+      return arrayAcc;
+    },
+    { add: {} }
+  );
+
+  return {
+    type: 'array',
+    label: field.label,
+    fields: arrayFields
+  };
+
+}
+
+function handleArea (field) {
+  const widgets = Object.keys(field.options.widgets).reduce((widgetAcc, widgetCur) => {
+    if (widgetTypes[widgetCur]) {
+      const newWidgetName = widgetTypes[widgetCur];
+      widgetAcc[newWidgetName] = field.options.widgets[widgetCur];
+    }
+    return widgetAcc;
+  }, {});
+
+  return {
+    ...field,
+    options: {
+      widgets
+    }
+  };
+}
+
+function handleRelationship(field) {
+  let max, builders;
+  const type = relationshipTypes[field.type];
+
+  if (field.type.includes('joinByOne')) {
+    max = 1;
+  }
+
+  if (field.filters) {
+    builders = field.filters;
+
+    if (field.filters.projection) {
+      builders.project = field.filters.projection;
+      builders.projection = undefined;
+    }
+
+    field.filters = undefined;
```

Comment:
will setting `filter` and `builders` to `undefined` remove them in schema.js?
otherwise we could just delete them

Reply:
Yes, it removes them